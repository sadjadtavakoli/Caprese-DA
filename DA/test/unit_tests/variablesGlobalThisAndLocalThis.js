this.variable = 1

function big(){
    function caller1(){
        let var1 = this.variable
        this.variable = 5
    }
    
    caller1()
    
}

function big2(){

    function caller2(){
        let var1 = this.variable
        this.variable = 5
    }
    
    function caller3(){
        let var1 = variable
        variable = 5
    }
    
    caller2()
    caller3()
}

big()
big2()

// caller1 impacts caller3 and caller2
// caller2 impacts caller3
