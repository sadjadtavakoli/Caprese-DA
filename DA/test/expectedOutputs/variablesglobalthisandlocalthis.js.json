{"big-test/unit_tests/variablesglobalthisandlocalthis.js-3-11":{"impacted":["t_1"]},"caller1-test/unit_tests/variablesglobalthisandlocalthis.js-4-7":{"impacted":["t_6","t_10","t_23","t_31"]},"big2-test/unit_tests/variablesglobalthisandlocalthis.js-13-27":{"impacted":["t_1"]},"caller2-test/unit_tests/variablesglobalthisandlocalthis.js-15-18":{"impacted":["t_18","t_23","t_31"]},"caller3-test/unit_tests/variablesglobalthisandlocalthis.js-20-23":{"impacted":["t_18","t_31"]},"keyMap":{"t_1":"test/unit_tests/variablesglobalthisandlocalthis.js","t_6":"big-test/unit_tests/variablesglobalthisandlocalthis.js-3-11","t_10":"caller1-test/unit_tests/variablesglobalthisandlocalthis.js-4-7","t_18":"big2-test/unit_tests/variablesglobalthisandlocalthis.js-13-27","t_23":"caller2-test/unit_tests/variablesglobalthisandlocalthis.js-15-18","t_31":"caller3-test/unit_tests/variablesglobalthisandlocalthis.js-20-23"}}